pkg/patterns/declarative/pkg/watch/dynamic.go

```diff
@@ -35,8 +36,19 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/log"
 )
 
-// WatchDelay is the time between a Watch being dropped and attempting to resume it
-const WatchDelay = 30 * time.Second
+var (
+	// WatchActivityTimeout sets a timeout for a Watch activity  under normal operation
+	WatchActivityTimeout = 300 * time.Second
```

Comment:
We _could_ have a method to enable the watch activity timeout, but I actually think it's a really good defensive feature so I think we should enable it by default anyway.  (If anyone disagrees, please comment!)

Reply:
ack